galgebra/metric.py

```diff
@@ -78,6 +78,22 @@ def linear_expand(expr, mode=True):
     else:
         return list(zip(coefs, bases))
 
+def collect(A, nc_lst):
+    """
+    A is a linear combination of noncommutative symbols with scalar
+    expressions as coefficients.  collect() takes the terms containing
+    the noncommutative symbols in nc_list and sums them so no elements
+    of nc_list appear more than once in the sum.  collect() combines all
+    coefficients of a given element of nc_lst into a single coefficient.
+    """
+    (coefs,bases) = linear_expand(A)
+    C = S(0)
+    for x in nc_lst:
+        if x in bases:
```

Comment:
What happens if `bases` contains elements not in `nc_list`?

Reply:
No, it's not SymPy's behavior:

![image](https://user-images.githubusercontent.com/64258/67020427-c9549a80-f130-11e9-918f-65339150c102.png)

Reply:
It will only collect for the symbols in `nc_list` and leave the rest unchanged.

Reply:
I think something as simple as the following would work?
```
def collect(x):
    return sum([coef*b for coef, b in zip(linear_expand(A))], start=S(0))
```

Reply:
Probably but I need to check the semantics of `linear_expand` that `collect` is making use of and add the doc for it as well.